button-merchant/src/main/java/com/usebutton/merchant/ConnectionManagerImpl.java

```diff
@@ -107,13 +109,16 @@ public NetworkResponse executeRequest(@NonNull ApiRequest request)
             if (responseCode >= 400) {
                 String message = "Unsuccessful Request. HTTP StatusCode: " + responseCode;
                 Log.e(TAG, message);
-                throw new ButtonNetworkException(message);
+                throw new HttpStatusException(message, responseCode);
             }
 
             JSONObject responseJson = readResponseBody(urlConnection);
             return new NetworkResponse(responseCode, responseJson);
-        } catch (IOException | JSONException e) {
-            Log.e(TAG, e.getClass().getSimpleName() + " has occurred", e);
+        } catch (IOException e) {
+            Log.e(TAG, "Error has occurred", e);
+            throw new NetworkNotFoundException(e);
```

Comment:
Are we sure there is no other reason an `IOException` would be thrown other than no network connection available?